### ISSUE TITLE
agent prevents ctrl-c from stopping cassandra

### ISSUE THREAD

**Issue by cscetbon** [instaclustr/cassandra-exporter#30]
If you use : 
- clone https://github.com/Orange-OpenSource/cassandra-image/tree/945a3ede7f8bd06c0f43ecedbc84bded38c6fea0
- apply patch https://gist.github.com/cscetbon/26f451c9b09480009a33bf59b6462d5f
- build the image using `make build`
- run the image using `docker run --name cassie --rm -ti -P --cap-add IPC_LOCK orangeopensource/cassandra-image:3.11.4-8u212-0.3.2`

and try to stop it using ctrl-c, the server does not stop completely. If you reset the last changes (`git reset --hard`), rebuild it, start it and try again, it works. It seems the exporter prevents the server from stopping when catching `SIGINT`.

**Comment by allamand** [comment]
Hi,

This becomes a real problem since once deployed in a Kubernetes cluster, this prevent the Pod to shutdown properly.

In my case, the Pod get stuck in `Terminating` state : 

```
NAME                        READY   STATUS        RESTARTS   AGE
cassandra-e2e-dc1-rack1-0   0/1     Terminating   0          11m
```

here is last logs on my pod:

```
**disablegossip**
WARN  [RMI TCP Connection(8)-172.17.0.6] 2019-09-05 13:10:16,409 StorageService.java:320 - Stopping gossip by operator request
INFO  [RMI TCP Connection(8)-172.17.0.6] 2019-09-05 13:10:16,412 Gossiper.java:1551 - Announcing shutdown
INFO  [RMI TCP Connection(8)-172.17.0.6] 2019-09-05 13:10:16,421 StorageService.java:2327 - Node /172.17.0.6 state jump to shutdown
**disablebinary**
INFO  [RMI TCP Connection(12)-172.17.0.6] 2019-09-05 13:10:29,501 Server.java:176 - Stop listening for CQL clients
**drain**
INFO  [RMI TCP Connection(16)-172.17.0.6] 2019-09-05 13:10:41,605 StorageService.java:1483 - DRAINING: starting drain process
INFO  [RMI TCP Connection(16)-172.17.0.6] 2019-09-05 13:10:41,608 HintsService.java:209 - Paused hints dispatch
INFO  [RMI TCP Connection(16)-172.17.0.6] 2019-09-05 13:10:41,705 Gossiper.java:1551 - Announcing shutdown
INFO  [RMI TCP Connection(16)-172.17.0.6] 2019-09-05 13:10:41,715 StorageService.java:2327 - Node /172.17.0.6 state jump to shutdown
INFO  [RMI TCP Connection(16)-172.17.0.6] 2019-09-05 13:10:43,717 MessagingService.java:981 - Waiting for messaging service to quiesce
INFO  [ACCEPT-/172.17.0.6] 2019-09-05 13:10:43,719 MessagingService.java:1336 - MessagingService has terminated the accept() thread
INFO  [RMI TCP Connection(16)-172.17.0.6] 2019-09-05 13:10:45,020 HintsService.java:209 - Paused hints dispatch
INFO  [RMI TCP Connection(16)-172.17.0.6] 2019-09-05 13:10:45,201 StorageService.java:1483 - DRAINED
**stop**
**stopdaemon**
INFO  [RMI TCP Connection(26)-172.17.0.6] 2019-09-05 13:11:21,313 CassandraDaemon.java:568 - Cassandra shutting down...
```

**Comment by zegelin** [comment]
Hi @cscetbon and @allamand. Thanks for the bug report. Agree that this is not ideal. I thought everything was running as daemon threads, which shouldn't prevent shutdown, but I must have missed something.

I have a bit of bandwidth this week to look into this further.

**Comment by cscetbon** [comment]
@zegelin that'd be great. Thank you

**Comment by zegelin** [comment]
This has been fixed in v0.9.8. Turns out we had added an explicit Netty server shutdown call in a previous release, and this was incorrectly trying to sync on a future.

**Comment by allamand** [comment]
Thanks a lot @zegelin, this solves our problem ;)
